### DISCUSSION TITLE
How to make onSuccess calback run every mount when useQuery refetchOnMount set to false?

### DISCUSSION THREAD

**Discussion by artidataio** [TanStack/query#3759] [Q&A] [answered]
Hi, I do not want **useQuery** to refetch on every mount, so I set the `refetchOnMount:false`. However, it seems that **useQuery's** `onSuccess` callback doesn't fire when it doesn't refetch. This is not the behaviour I want, I still want the `onSuccess` callback to fire on the _first successful fetch_, and it also fires on every subsequent component remounting based on the _first successful fetch_ data, even though it doesn't refetch.  Can that behaviour be achieved?

**Comment by TkDodo** [comment, marked as answer]
onSuccess is tied to successful fetches. No fetch - no callback. What are you trying to do in that callback?

**Reply by artidataio** [reply]
Figured out a way not to use `onSuccess` callback after all. Thanks!
